pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/properties/ValueParsers.java

```diff
@@ -0,0 +1,254 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.rule.properties;
+
+import static net.sourceforge.pmd.lang.rule.properties.modules.MethodPropertyModule.ARRAY_FLAG;
+import static net.sourceforge.pmd.lang.rule.properties.modules.MethodPropertyModule.CLASS_METHOD_DELIMITER;
+import static net.sourceforge.pmd.lang.rule.properties.modules.MethodPropertyModule.METHOD_ARG_DELIMITER;
+import static net.sourceforge.pmd.lang.rule.properties.modules.MethodPropertyModule.METHOD_GROUP_DELIMITERS;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+
+import net.sourceforge.pmd.util.ClassUtil;
+
+/**
+ * @author Clément Fournier
+ * @since 6.0.0
+ */
+//TODO: make package-private when we move everything to n.s.pmd.properties
+public final class ValueParsers {
```

Comment:
I like the idea of moving these out, but I believe we need a different name.

In general, it's a a code smell to have classes differing on a single char, specially plurals

Reply:
Maybe `ValueParserConstants` then ?

Reply:
That's a good name!